### ISSUE TITLE
Test error handling for invalid LF programs

### ISSUE THREAD

**Issue by cmnrd** [icyphy/lingua-franca#461]
It seems that there are no tests that use an invalid input to test the error handling mechanism of our compiler. This leads to bugs like #460, and I suspect that there are more similar problems. I think there should be a testing infrastructure that is focused on invalid programs and that checks if errors are reported correctly.

**Comment by oowekyala** [comment]
I think I can help with this, I'll see what I can do

**Comment by lhstrh** [comment]
I think that the work done by @petervdonovan on error handling and error reporting tests in the LSP server already address this issue. Can we close this?

**Comment by petervdonovan** [comment]
> I think that the work done by @petervdonovan on error handling and error reporting tests in the LSP server already address this issue.

Those tests only check invalid target code, so I doubt they can help with problems like #460.

**Comment by lhstrh** [comment]
I don't think the point here is that those tests help with problems like #460. That issue describes a programming error that lead to a null pointer exception. This problem was fixed, but this issue reflects on our ability to test that such a faulty LF program triggers the appropriate error response from the compiler. I think that part of this is covered by the unit tests; another part of this is covered by the LSP tests. I think it's difficult to come up with any comprehensive strategy, however, so I'm inclined to say that we've made sufficient progress toward the stated goal to where I think we can close this.

**Comment by cmnrd** [comment]
I agree that there is no ultimate fix to this issue. However, we should probably try to be more disciplined about testing and maybe should add more rules (or recommendations) in CONTRIBUTING.md. For every single validation error there should be at least one unit test making sure that the error is reported (and nothing else goes wrong along the way).
